Sources/CoreMetrics/Metrics.swift

```diff
@@ -39,10 +42,32 @@ public final class Counter {
     ///     - label: The label for the `Counter`.
     ///     - dimensions: The dimensions for the `Counter`.
     ///     - handler: The custom backend.
-    public init(label: String, dimensions: [(String, String)], handler: CounterHandler) {
+    ///     - factory: The custom factory.
+    public init(label: String, dimensions: [(String, String)], handler: CounterHandler, factory: MetricsFactory) {
         self.label = label
         self.dimensions = dimensions
         self._handler = handler
+        self._factory = factory
+    }
+
+    /// Alternative way to create a new `Counter`, while providing an explicit `CounterHandler`.
+    ///
+    /// - warning: This initializer provides an escape hatch for situations where one must use a custom factory instead of the global one.
+    ///            We do not expect this API to be used in normal circumstances, so if you find yourself using it make sure it's for a good reason.
```

Comment:
Is this doc comment correct here (Same for the other types)? This init doesn't take a factory so no way to pass a custom factory. I also don't think we need this and both inits are totally valid to use.

Reply:
The handler is the "factory" here, I just copied the comment from the existing method.

If we want to clean it up, we should do it separately.